### DISCUSSION TITLE
Support Comments

### DISCUSSION THREAD

**Discussion by TonyGravagno** [araera111/mysql-to-zod#29] [Ideas]
If the SQL includes comments, I'd like to add it into the resulting .ts : 

```sql
  `last_modified_by` bigint(20) unsigned NOT NULL
         COMMENT 'ID links to person',
  `created_time` timestamp(4) NOT NULL DEFAULT current_timestamp(4)
         COMMENT 'Only set on initial save',
  `last_modified_time` timestamp(4) NOT NULL DEFAULT current_timestamp(4)
         COMMENT 'Always modified',
```

Default code style, maybe configurable as seen below :  

```ts
  // lastModifiedBy : ID links to person
  lastModifiedBy: z.number().nonnegative(),
  // createdTime : Only set on initial save
  createdTime: z.date(),
  // lastModifiedTime : Always modified
  lastModifiedTime: z.string(),
```
Alternate style 1 - choice may depend on expected length

```ts
  lastModifiedBy: z.number().nonnegative(), // ID links to person
  createdTime: z.date(), // Only set on initial save
  lastModifiedTime: z.string(), // Always modified
```

Alternate style 2 - Long comments always go on separate line

```ts
  lastModifiedBy: z.number().nonnegative(), // ID links to person
  // Only set on initial save using UTC, do not insert/update this field
  createdTime: z.date(),
  lastModifiedTime: z.string(), // Always modified
```

**Comment by TonyGravagno** [comment]
There is a better way to do this, noted in [#34](https://github.com/araera111/mysql-to-zod/discussions/34#discussioncomment-6569681). I don't like my idea about putting a comment on a code line because schema is very often multi-line and comments on the `z.` line would look weird. I think it's better and easier to simply have all comments above their code line.

So the suggestion here is still valid for field-level comments, but implemented in exactly the same way as tables as discussed in #34.

**Comment by araera111** [comment]
to issue.
